market/ipni/spark/sol/README.md

```diff
@@ -0,0 +1,302 @@
+# **MinerPeerIDMapping Smart Contract**
+
+## **Overview**
+
+The `MinerPeerIDMapping` contract provides a decentralized way to map Filecoin miner IDs (`uint64`) to their PeerIDs along with signed messages for validation. It ensures only authorized entities (owner or controlling addresses) can add, update, or delete mappings.
```

Comment:
Should this line also mention that the smart contract collects data necessary to verify that the miner is also controlling the PeerID submitted to the mapping table?

Reply:
Signed message line says the same. Any public signature byte can be verified.

Reply:
I see, you are referring to this line:

https://github.com/filecoin-project/curio/pull/377/files/dce763f30955515840aa032240065638e3a8dcbc#diff-8ee901221d95fab247080ab3104e7497e45f59edc37e7b8d9d14f2811fdc03f4R130

> This message is signed using the private key associated with the on-chain PeerID.

As I am trying to explain in https://github.com/filecoin-project/curio/pull/377#discussion_r1937726379 and https://github.com/filecoin-project/curio/pull/377#discussion_r1954671775, I think this is a mistake - the message should be signed using the private key associated with with the PeerID in the message.